illumina.py

```diff
@@ -520,10 +522,14 @@ def get_fname(self):
 
     def get_flowcell(self):
         fc = self.root[0].find('Flowcell').text
-        if '-' in fc:
-            # miseq often adds a bunch of leading zeros and a dash in front
-            fc = fc.split('-')[1]
-        assert 4 <= len(fc) <= 9
+        # do not slice in the iSeq/Firefly case
+        if not self.get_machine().startswith("FF"):
+            if '-' in fc:
+                # miseq often adds a bunch of leading zeros and a dash in front
+                fc = fc.split('-')[1]
+        # >=5 to avoid an exception here: https://github.com/broadinstitute/picard/blob/2.17.6/src/main/java/picard/illumina/IlluminaBasecallsToSam.java#L510
+        # <= 15 to limit the bytes added to each bam record
```

Comment:
I'm not sure that we have to enforce an upper bound for the sake of reducing BAM file size (Picard already truncates this anyway), but it's probably good to limit it anyway, or at least bring it to our attention if they end up longer than we thought.